10 missing on air functionality

helo-head opened the pull request:
Should be ready for review

helo-head committed:
Beta "On Air" Functionality

Beta of new "On Air" functionality using modified ATEMstd library

helo-head committed:
Updated Documentation

brianteeman commented on Tallly-Core2-V11.ino:
```diff
@@ -48,14 +49,17 @@ boolean writeEEconfig();
 #define BS_BYLOC 0    // 109 Second Row : 0 First Row
 #define BS_BXLOC 265  // 285 Far Right : 265 Forth Column
+#define OAS_XLOC 135  // Right of center second row
+#define OAS_YLOC 0    // First Row
```
```suggestion
#define OAS_YLOC 7    // First Row
```

brianteeman commented on Tallly-Core2-V11.ino:
```diff
@@ -48,14 +49,17 @@ boolean writeEEconfig();
 #define BS_BXLOC 265  // 285 Far Right : 265 Forth Column
+#define OAS_XLOC 135  // Right of center second row
+#define OAS_YLOC 0    // First Row
+#define OAS_SIZE 7    // Standard Size
```
```suggestion
#define OAS_SIZE 6    // Standard Size
```

brianteeman commented on Tallly-Core2-V11.ino:
```diff
@@ -1151,7 +1162,21 @@ void loop() {
+    M5.Lcd.fillCircle(OAS_XLOC, OAS_YLOC, OAS_SIZE, TFT_LIGHTGREY);    // If ATEM not connected then "On Air" is unknown at best
+    atemOAStatus = false;
+    atemLastOAStatus = false;
+    }  // End "Aon Air" status check
```
```suggestion
    }  // End "On Air" status check
```

brianteeman commented:
Done a quick review to test compiling and default status
I assume that it is supposed to be a full circle and not a half circle so changed the LOC values

Will test the actual onair status is working in a few hours when I have a bit more time.

helo-head committed:
Update Tallly-Core2-V11.ino

Co-authored-by: Brian Teeman <brian@teeman.net>

helo-head committed:
Update Tallly-Core2-V11.ino

Co-authored-by: Brian Teeman <brian@teeman.net>

helo-head committed:
Update Tallly-Core2-V11.ino

Co-authored-by: Brian Teeman <brian@teeman.net>

helo-head committed:
Fixed ACS location to match documentation

brianteeman commented on Tallly-Core2-V11.ino:
```diff
@@ -1151,7 +1162,21 @@ void loop() {
+    atemOAStatus = AtemSwitcher.getStreamStreaming();
+    if (atemOAStatus != atemLastOAStatus) {
+      atemLastOAStatus = atemOAStatus;
+      M5.Lcd.fillCircle(ACS_XLOC, ACS_YLOC, ACS_SIZE, atemOAStatus ? TFT_RED : TFT_LIGHTGREY);  // ATEM "On Air" Status
```
```suggestion
      M5.Lcd.fillCircle(OAS_XLOC, OAS_YLOC, OAS_SIZE, atemOAStatus ? TFT_RED : TFT_LIGHTGREY);  // ATEM "On Air" Status
```

I think you made a copy paste error here. as when you are on air you are changing the color of the connection icon.

helo-head committed:
Update Tallly-Core2-V11.ino

Co-authored-by: Brian Teeman <brian@teeman.net>

helo-head commented:
Assuming the streaming/OnAir code works I need to add one more small bit of code to cover the case where someone wants to run this on a mini that does not support streaming.  Based on my current track record you will probably need to test it to make sure I don't break anything :)

brianteeman commented:
> Assuming the streaming/OnAir code works

sorry forgot to say that it worked

helo-head committed:
OAS Conditional Support

Changed OAS code to be conditional based on definition of OAS_SUPP

brianteeman commented:
am I correct that the condition is not dynamic but by manually changing the value of #define OAS_SUPP true 

If that is correct then the code looks ok and you just need to mention how to disable the ON AIR in the readme

helo-head commented:
Added conditional OAS support, updated readme documentation to reflect Mini Pro+ requirement for OAS. 

Im debating the wisdom of changing filename and code in the same commit as you loose visibility into the code change. Not even sure a filename change is needed but originally did it to provide clarity.

helo-head committed:
Updated OAS documenation

helo-head commented:
Hopefully its clear how to enable ... Im going to leave it enabled by default since the Mini (no pro) is no longer being sold.
